string name
---
float32 real_x
float32 real_y